webcam-capture-drivers/driver-screencapture/src/main/java/com/github/sarxos/webcam/ds/gstreamer/ScreenCaptureDevice.java

```diff
@@ -59,14 +62,25 @@ public Dimension getResolution() {
 
 	@Override
 	public void setResolution(Dimension size) {
-		// do nothings, screen has only one resolution which is already set
+	    resolution.setSize(size.getWidth(), size.getHeight());
 	}
 
 	@Override
 	public BufferedImage getImage() {
 		final GraphicsConfiguration gc = device.getDefaultConfiguration();
 		final Rectangle bounds = gc.getBounds();
-		return robot.createScreenCapture(bounds);
+		BufferedImage screen = robot.createScreenCapture(bounds);
+		int width = resolution.width;
+		int height = resolution.height;
+		BufferedImage img = new BufferedImage(width, height, screen.getType());
```

Comment:
Please consider checking if screen image from Robot has the same size as resolution Dimension and return original screen image in such a case, so we do not need to create resized image when it's unnecessary.

```java
if (screen.getWidth() == width && screen.getHeight() == height) {
  return screen;
}
```